### PR TITLE
Fixed 'Mean of empty slice' warning

### PR DESCRIPTION
**Pull request by carefree0910**
## Motivation

When using `optuna`'s pruning strategy with `n_jobs > 1`, the following warning happens to occur frequently:

```python
...\site-packages\numpy\lib\nanfunctions.py:1366: RuntimeWarning:

Mean of empty slice
```

## Description of the changes

It's easy to be fixed with a simple condition statement. I've realized that you are using `black` to format your codes, so I've tried my best to follow it 😉

> BTW: I'm not sure whether you prefer returning `nan` of `inf` in such condition. I've chosen `inf` but feel free to change it to `nan` if you like 😆

### CODE REVIEW

**Comment by hvy** [comment]
Thanks a lot for the PR, this indeed seems like a bug. 

We cannot in my understanding determine if the trial should be pruned if records are missing, so we should return `False` from `should_prune`. I think doing an early return with nan will simplify the code/make it less error prone.

**Comment by carefree0910** [comment]
Not sure if it requires me to edit it but I've made a commit anyway 🤣

File: optuna/pruners/_percentile.py [hunk 1]
```diff
@@ -33,19 +33,20 @@ def _get_percentile_intermediate_result_over_trials(
     if len(completed_trials) == 0:
         raise ValueError("No trials have been completed.")
 
-    if direction == StudyDirection.MAXIMIZE:
+    intermediate_values = [
+        t.intermediate_values[step] for t in completed_trials if step in t.intermediate_values
+    ]
+    is_maximize = direction == StudyDirection.MAXIMIZE
```
Comment (hvy) [inline optuna/pruners/_percentile.py:39]: This is a small detail but this local variable doesn't really improve the code quality and it's somewhat unrelated to the fix. Might it be sensible to revert this change?

**Review by hvy — COMMENTED** [review]
Thanks again, the changes basically LGTM. I just left one nitpick.

File: optuna/pruners/_percentile.py (hunk 1 above)
Comment (carefree0910) [reply optuna/pruners/_percentile.py]: Sure! I forgot to revert this after I returned `math.nan` 🤣

**Review by hvy — APPROVED** [review]
LGTM!

**Review by HideakiImamura — APPROVED** [review]
Thank you for the update. LGTM!
